woocommerce/woocommerce-ios
# Finalize Analytics Usage Tracking

This enhances #5503 so that it can be merged. I created a separate PR so that it's hopefully easier to review. 

## Changes

### Removed Singleton

I removed the `StoreStatsUsageTracksEventEmitter` singleton and replaced it with injected properties instead: 5a834ce19ddde1a93724693a6dc860662a9bb16b. 

In relation to this, I changed the My Store tab press detection to use `viewWillAppear` in b48094f27ef22a417e76d8c16a689350f5742bed to keep the `StoreStatsUsageTracksEventEmitter` instance within the `StoreStats*` classes. It felt incorrect to add the `StoreStatsUsageTracksEventEmitter` inside `MainTabBarController`. And subsequently, I realized that this causes duplication of "interaction tracking" since [`StoreStatsV4PeriodViewController. trackChangedTabIfNeeded`](https://github.com/woocommerce/woocommerce-ios/blob/issue/5501-analytics-tracking-finalize/WooCommerce/Classes/ViewRelated/Dashboard/Stats%20v4/StoreStatsV4PeriodViewController.swift#L581) is also executed when the merchant navigates back to the My Store tab. So I just removed it in 0316ac3ed201166ee76b6ecd2c47456fb0d47c86. 

### Added Tracking to Old VCs

I noticed that there are `OldStoreStats*` classes now. I made sure to include the same interaction tracking to these classes. 

- 8cab99044321f277b46493fbfc43bb48abc8c4d3
- ce0cfb21abf48f58e99e84dc1cd8a9abc1e6f708
- 197086b4f4680e4af74b9f4f523b220c4dd1c2e6

### Others

- I removed the notice call in 4d75603b174a0452e9ccdf03411dec2360b9f405. That was mainly used for testing and playing with this algorithm. 
- I applied the suggested changes from the #5503 PR: c83325ef6eb0cb12b35308cbb40448327fe17ac2 and ea5fa60ea597affb0dc0f4fe02cafaeab1debac2. 

## Testing

1. Log in
2. Navigate to My Store
3. Do a few of the following:
     - Scrolling
     - Pull-to-refresh
     - Tapping on the bars in the chart
     - Changing the tab
     - Navigating to the My Store tab
     - Tapping on a product in the Top Performers list
4. Confirm that after performing at least 5 of the interactions within more than 10 seconds, a `used_analytics` Tracks event is logged.
5. Turn off the [`myStoreTabUpdates`](https://github.com/woocommerce/woocommerce-ios/blob/b17f04c03fa6e882122dd170cfa548837d7b245a/Experiments/Experiments/DefaultFeatureFlagService.swift#L36-L37) feature flag and repeat steps 3 and 4. 


## Author Checklist
